src/dynamics/mpas/driver/dyn_mpas_subdriver.F90

```diff
@@ -2776,6 +2777,106 @@ subroutine dyn_mpas_compute_edge_wind(self, wind_tendency)
         call self % debug_print(log_level_debug, subname // ' completed')
     end subroutine dyn_mpas_compute_edge_wind
 
+    !-------------------------------------------------------------------------------
+    ! subroutine dyn_mpas_compute_cell_relative_vorticity
+    !
+    !> summary: Compute the relative vorticities at cell points.
+    !> author: Kuan-Chih Wang
+    !> date: 2025-04-12
+    !>
+    !> MPAS uses staggered C-grid for spatial discretization, where relative
+    !> vorticities are located at vertex points because wind vectors are located at
+    !> edge points. However, physics schemes that use relative vorticities as input
+    !> usually want them at cell points instead.
+    !> This subroutine computes the relative vorticity at each cell point from its
+    !> surrounding vertex points and returns the results.
+    !
+    !-------------------------------------------------------------------------------
+    subroutine dyn_mpas_compute_cell_relative_vorticity(self, cell_relative_vorticity)
+        class(mpas_dynamical_core_type), intent(in) :: self
+        real(rkind), allocatable, intent(out) :: cell_relative_vorticity(:, :)
+
+        character(*), parameter :: subname = 'dyn_mpas_subdriver::dyn_mpas_compute_cell_relative_vorticity'
+        integer :: i, k
+        integer :: ierr
+        integer, pointer :: ncellssolve, nvertlevels
+        integer, pointer :: kiteforcell(:, :), nedgesoncell(:), verticesoncell(:, :)
+        real(rkind), pointer :: areacell(:), kiteareasonvertex(:, :), vorticity(:, :)
+
+        nullify(ncellssolve, nvertlevels)
+        nullify(kiteforcell, nedgesoncell, verticesoncell)
+        nullify(areacell, kiteareasonvertex, vorticity)
+
+        ! Input.
+        call self % get_variable_pointer(ncellssolve, 'dim', 'nCellsSolve')
+        call self % get_variable_pointer(nvertlevels, 'dim', 'nVertLevels')
+
+        call self % get_variable_pointer(kiteforcell, 'mesh', 'kiteForCell')
+        call self % get_variable_pointer(nedgesoncell, 'mesh', 'nEdgesOnCell')
+        call self % get_variable_pointer(verticesoncell, 'mesh', 'verticesOnCell')
+
+        call self % get_variable_pointer(areacell, 'mesh', 'areaCell')
+        call self % get_variable_pointer(kiteareasonvertex, 'mesh', 'kiteAreasOnVertex')
+        call self % get_variable_pointer(vorticity, 'diag', 'vorticity')
+
+        ! Output.
+        allocate(cell_relative_vorticity(nvertlevels, ncellssolve), stat=ierr)
```

Comment:
Just an FYI that we have started trying to provide the Fortran-provided error message from allocate statements as well, e.g.:

```
character(256) :: errmsg
...
allocate(cell_relative_vorticity(nvertlevels, ncellssolve), stat=ierr, errmsg=errmsg)
```

Of course this change certainly isn't required for this PR.